subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/PendingDependenciesState.java

```diff
@@ -26,18 +25,29 @@
 public class PendingDependenciesState {
     private static final PendingDependencies NOT_PENDING = PendingDependencies.notPending();
 
-    private final Map<ModuleIdentifier, PendingDependencies> pendingDependencies = Maps.newHashMap();
+    private final Map<String, Map<String, PendingDependencies>> pendingDependencies = Maps.newHashMap();
```

Comment:
If we can streamline getting the `ModuleIdentifier`, we can avoid all of these Maps...